app/controllers/islands_controller.rb

```diff
@@ -6,4 +6,20 @@ def index
   def show
     @island = Island.find(params[:id])
   end
+
+  def edit
+    @island = Island.find(params[:id])
+  end
+
+  def update
+    @island = Island.find(params[:id])
+    @island.update(params[:island])
+    redirect_to_island_path(@island)
+  end
+end
+
+private
+
```

Comment:
lines 7, 11  and 15 could be refactored here : 

def set_island
   @island = Island.find(params[:id])
end